packages/cdktf/lib/terraform-functions.ts

```diff
@@ -0,0 +1,1221 @@
+import { Tokenization } from "./tokens/token";
+import { call } from "./tfExpression";
+import { IResolvable } from "./tokens/resolvable";
+import { TokenMap } from "./tokens/private/token-map";
+
+// We use branding here to ensure we internally only handle validated values
+// this allows us to catch usage errors before terraform does in some cases
+type TFValue = any & { __type: "tfvalue" };
+type TFValueValidator = (value: any) => TFValue;
+
+type ExecutableTfFunction = (...args: any[]) => IResolvable;
+
+function anyValue(value: any): any {
```

Comment:
Maybe somehow group these validators to make it more clear that they are validators and not just arbitrary functions